packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManager.java

```diff
@@ -389,6 +391,47 @@ public static long measureText(
             ? paragraphAttributes.getInt(MAXIMUM_NUMBER_OF_LINES_KEY)
             : UNSET;
 
+    int numberOfLines =
+      paragraphAttributes.hasKey(NUMBER_OF_LINES_KEY)
+        ? paragraphAttributes.getInt(NUMBER_OF_LINES_KEY)
+        : UNSET;
+
+    int lines = layout.getLineCount();
+    if (numberOfLines != UNSET && numberOfLines != 0 && numberOfLines >= lines && text.length() > 0) {
+      int numberOfEmptyLines = numberOfLines - lines;
+      SpannableStringBuilder ssb = new SpannableStringBuilder();
+
+      // for some reason a newline on end causes issues with computing height so we add a character
+      if (text.toString().endsWith("\n")) {
+        ssb.append("A");
+      }
```

Comment:
Could you elaborate on this? Do we see layout issues now when we just have new lines, or does the underlying spannable have some other form we can replicate?

Reply:
I had issues when only new lines were used. There was a weird jump after someone added 2 newlines in a row. It just ignored the second one. This fixed it. It's not visible on the screen as it's only used in layout phase.